src/epics/revisions.js

```diff
@@ -1,68 +1,254 @@
-										id: data.revid,
-										...data
-									} )
-								]
-							) )
-						)
-							// Since the ids are all from the same wiki, they are in a
-							// guaranteed order from oldest to newest.
-							.sort( ( a, b ) => a.id - b.id );
-
-					return RevisionsActions.setRevisions( revisions );
+		.flatMap( ( action ) => {
+			// Remove any users who have a false continue.
+			const users = action.users.filter( ( user ) => {
+				return store.getState().revisions.cont.get( user ) !== false;
+			} );
+
+			// If the set of users is empty, no need to attempt a request.
+			if ( users.isEmpty() ) {
+				return Observable.of( [] );
+			}
+
+			const requests = users
+				.map( user => {
+					// Set the variables so the request can be canceled if the state
+					// changes.
+					const wiki = store.getState().query.wiki;
+					const domain = store.getState().wikis.get( wiki ).domain;
+					const startDate = store.getState().query.startDate;
+					const endDate = store.getState().query.endDate;
+					const cont = store.getState().revisions.cont.get( user );
+
+					return Observable.ajax( {
+						url: buildRevisionUrl( domain, user, startDate, endDate, cont ),
+						crossDomain: true,
+						responseType: 'json'
+					} )
+						.map( ( ajaxResponse ) => {
+							const contribs = ajaxResponse.response.query ? ajaxResponse.response.query.usercontribs : [];
+
+							const revisions = new OrderedMap(
+								contribs.map( ( data ) => (
+									[
+										data.revid,
+										new Revision( {
+											id: data.revid,
+											...data
+										} )
+									]
+								) )
+							);
+
+							return {
+								revisions,
+								cont: new Map( {
+									[ user ]: ajaxResponse.response.continue ? ajaxResponse.response.continue.uccontinue : false
+								} )
+							};
+						} )
+						// If the query is no longer valid, cancel the request.
+						.takeUntil( action$.ofType( 'REVISIONS_NOT_READY' ) )
+						// If the user is no longer in the query, cancel the request.
+						.takeUntil( action$.ofType( 'QUERY_USER_CHANGE' ).filter( a => !a.users.includes( user ) ) )
+						// If the wiki changes, cancel the request.
+						.takeUntil( action$.ofType( 'QUERY_WIKI_CHANGE' ).filter( a => a.wiki !== wiki ) )
+						// If the start date changes, cancel the request.
+						.takeUntil( action$.ofType( 'QUERY_START_DATE_CHANGE' ).filter( a => a.startDate !== startDate ) )
+						// If the end date changes, cancel the request.
+						.takeUntil( action$.ofType( 'QUERY_END_DATE_CHANGE' ).filter( a => a.endDate !== endDate ) );
+				} );
+
+			return Observable.forkJoin( requests.toArray() );
+		} )
+		.flatMap( ( data ) => {
+
+			if ( data.length === 0 ) {
+				return Observable.of( RevisionsActions.addRevisions( new OrderedMap() ) );
+			}
+
+			const revisions = data.reduce( ( map, item ) => {
+				return map.merge( item.revisions );
+			}, new OrderedMap() );
+			const cont = data.reduce( ( map, item ) => {
```

Comment:
`cont` is not very descriptive and it is used all over the place. there is algo a reducer call cont.js. I think this should be renamed to make it more readable.

Reply:
@dayllanmaza What would you prefer? it's short for `continue` but that's a protected word.

Reply:
For lack of a better name, lets leave it like that.